src/main/kotlin/org/jitsi/nlj/RtpSenderImpl.kt

```diff
@@ -216,7 +219,12 @@ class RtpSenderImpl(
             numQueueReads++
             lastQueueReadTime = now
             packetInfo.addEvent(PACKET_QUEUE_EXIT_EVENT)
-            outgoingRtpRoot.processPacket(packetInfo)
+
+            val root = when (packetInfo.packet) {
```

Comment:
no data to back it up, but given this is in such a hot path i worry this is extra cycles where the old code wasn't really that bad: we knew when we had rtp and rtcp so we sent them directly, but now we merge them together just so they can be split up again.

Reply:
What would the alternative be? Add a second queue (with its own thread) for RTCP? And all we would gain is avoiding a single 'instanceof' check for each packet. We could add an RTP/RTCP flag to PacketInfo, but I don't think that would be much faster than 'instanceof'

If I am reading the code correctly we are currently fanning out SRs and passing them through all send pipelines in the receiver thread (the same one which runs RTP!).

Reply:
i agree with boris that we probably shouldn't worry too much about the performance impact of the additional instanceof check, but it also kind of bothers me to have to merge rtp/rtcp together so they can be split up again.. seems like something is conceptually wrong. would it be possible to avoid splitting in the first place, i.e. remove the sendRtcp method and only keep sendPacket? so we split at one place

Reply:
Yes, this sounds good. It will also avoid re-creating the PacketInfo instance

Reply:
I'll do that in a separate PR